x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/execution/search/extractor/FieldHitExtractor.java

```diff
@@ -127,8 +127,39 @@ private Object unwrapMultiValue(Object values) {
         if (values == null) {
             return null;
         }
-        if (dataType == DataType.GEO_POINT || dataType == DataType.GEO_SHAPE) {
-            return extractGeoShape(values);
+        if (dataType == DataType.GEO_POINT) {
+            Object value;
+            if (values instanceof List) {
+                List<?> list = (List<?>) values;
+                if (list.isEmpty()) {
+                    // we get geo_points from doc values and they were not specified for this record
+                    return null;
+                }
+                if (((List<?>) values).size() == 1) {
+                    value = ((List<?>) values).get(0);
+                } else {
+                    // This could be a single point in [lon, lat] format or multiple points
+                    if ((list.size() == 2 || list.size() == 3) && (list.get(0) instanceof Number)) {
```

Comment:
It would be good if you can add a comment explaining the `size() == 3` scenario when this can happen. Something like `// This could be a single point in [lon, lat(, altitude)] (where altitude is optional) format or multiple points`